src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp

```diff
@@ -908,16 +908,17 @@ namespace
                     precomputed_mis_weights = scattering_coef;
                 else
                     Spectrum::upgrade(scattering_coef, precomputed_mis_weights);
-                precomputed_mis_weights *= vertex.m_throughput;
+                if (vertex.m_throughput.size() <= channel_count)
```

Comment:
It is wrong, you are right. It is the result of my laziness: the fact is if we upgrade everything to the max size, we need to tell `PhaseFunction::sample_distance` what spectrum size we are using. Changing `PhaseFunction::sample_distance` is starting to get annoying... Possibly, we should remove this functionality from `PhaseFunction` forever and ever.